config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -180,4 +180,34 @@ private ContextSpecificConfig getContextSpecificConfig(ConfigDocument configDocu
         .setUpdateTimestamp(configDocument.getUpdateTimestamp())
         .build();
   }
+
+  private ContextSpecificConfig getContextSpecificConfig(
+      ConfigDocument configDocument, ContextSpecificConfig existingConfig) {
+    if (ConfigServiceUtils.isNull(existingConfig.getConfig())) {
+      return getContextSpecificConfig(configDocument);
+    } else {
+      return ContextSpecificConfig.newBuilder()
+          .setConfig(configDocument.getConfig())
+          .setContext(configDocument.getContext())
+          .setCreationTimestamp(configDocument.getCreationTimestamp())
+          .setUpdateTimestamp(configDocument.getUpdateTimestamp())
+          .setPrevConfig(existingConfig.getConfig())
```

Comment:
Ah - so this is why this got added. We should find another way to return this such as a wrapper, because this field doesn't make sense on the api, only for internal usage.